packages/mds-compliance-engine/engine/helpers.ts

```diff
@@ -69,7 +80,66 @@ export function isPolicyActive(policy: PolicyDomainModel, end_time: number = now
   return end_time >= policy.start_date && end_time <= policy.end_date
 }
 
-export function isRuleActive(rule: Rule<Exclude<RULE_TYPE, 'rate'>>): boolean {
+/**
+ * Luxon has a helper method for this `weekdayShort`, but it doesn't typecheck :^)
+ */
+const numericalWeekdayToLocale = (weekdayNum: number) => {
+  if (weekdayNum < 1 || weekdayNum > 7) {
+    throw new Error(`Invalid weekday number: ${weekdayNum}`)
+  }
+
+  const weekdayList = <const>['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun']
+
+  return weekdayList[weekdayNum - 1] // subtract 1 cause arrays are 0 indexed, but luxon provides weekdays in 1 | 2 | 3 | 4 | 5 | 6 | 7 form. (1 = monday) (7 = sunday)
+}
+
+/**
+ *
+ * @param param0 Object which contains a list of days
+ * @returns If the current day is in the list of valid days
+ */
+const isCurrentDayInDays = ({ days }: Pick<Rule, 'days'>) => {
+  const { TIMEZONE } = env
+
+  if (!days || days.length === 0) return true
+
+  const currentTime = DateTime.now().setZone(TIMEZONE)
+
+  return days.includes(numericalWeekdayToLocale(currentTime.weekday))
+}
+
+/**
+ *
+ * @param start_time Time in HH:mm:ss format
+ * @param end_time Time in HH:mm:ss format
+ * @returns If the current local time is within the specified time range.
+ *
+ * Note: We can just do string comparison here for times, because the format should consistently be HH:mm:ss for all strings.
+ */
+const isCurrentTimeInInterval = ({ start_time, end_time }: Pick<Rule, 'start_time' | 'end_time'>) => {
+  const { TIMEZONE } = env
+
+  const currentTime = DateTime.now().setZone(TIMEZONE)
+  const formattedCurrentTime = currentTime.toFormat(TIME_FORMAT)
+  if (!isOfTimeFormat(formattedCurrentTime))
```

Comment:
Is this really necessary? You just formatted the current time yourself using an internally determined format, so how would this ever throw?

Reply:
Yeah, this should basically never throw, and was added for the sake of type safety without using an `as` assertion. The return type from `toFormat` is just a `string`, and because the template literal type `${number}:${number}:${number}` is not necessarily derivable from `string`, we need to either assert the type, or write a typeguard. I figured the extra sub-ms of runtime using the typeguard was nicer than an assertion, but totally up for changing it to an assertion because we're basically guaranteed that this conversion succeeds (unless there's a regression in the `luxon` library, which is unlikely).

Reply:
Ah. Maybe leave a comment that this is essentially a type guard, then.